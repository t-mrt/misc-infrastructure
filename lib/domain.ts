import * as cdk from "@aws-cdk/core"
import * as r from "@aws-cdk/aws-route53"
import * as t from "@aws-cdk/aws-route53-targets"
import * as c from "@aws-cdk/aws-cloudfront"

// TODO: divide stack by hosted zone

export class DomainStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props)
        const zone = new r.PublicHostedZone(this, "yux3", {
            zoneName: "yux3.net",
        })
        new r.CnameRecord(this, "blog", {
            zone: zone,
            domainName: "hatenablog.com.",
            recordName: "blog",
        })
        const distribution = c.Distribution.fromDistributionAttributes(this, "ImportedDist", {
            domainName: "d25ymprel65g4i.cloudfront.net",
            distributionId: "E1AB8RFEZ1H7F0",
        })
        new r.ARecord(this, "img", {
            zone: zone,
            target: r.RecordTarget.fromAlias(new t.CloudFrontTarget(distribution)),
            recordName: "img",
        })
    }
}
