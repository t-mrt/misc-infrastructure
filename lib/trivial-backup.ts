import * as cdk from "@aws-cdk/core"
import * as iam from "@aws-cdk/aws-iam"
import * as s3 from "@aws-cdk/aws-s3"

export class TrivialBackupStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props)
        const user = new iam.User(this, "User", {
            userName: "trivial_backup",
        })

        const bucket = new s3.Bucket(this, "Bucket", {
            versioned: true,
            bucketName: "trivial-backup-02",
            blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
        })

        bucket.addToResourcePolicy(
            new iam.PolicyStatement({
                actions: ["s3:PutObject"],
                resources: [bucket.arnForObjects("*")],
                principals: [user],
            })
        )
    }
}
