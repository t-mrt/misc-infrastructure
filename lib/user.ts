import * as cdk from '@aws-cdk/core';
import * as iam from '@aws-cdk/aws-iam'

export class UserStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);
    const user = new iam.User(this, 'GeneralPurposeUser', {
      userName: "t_"
    });
    user.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName("AdministratorAccess"))
  }
}
