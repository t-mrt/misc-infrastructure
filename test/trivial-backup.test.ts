import * as cdk from "@aws-cdk/core"
import { SynthUtils } from "@aws-cdk/assert"
import * as t from "../lib/trivial-backup"

test("snapshot", () => {
    const app = new cdk.App()
    const stack = new t.TrivialBackupStack(app, "MyTestStack")
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
})
