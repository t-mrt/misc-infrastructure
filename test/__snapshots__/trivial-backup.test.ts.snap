// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "Resources": Object {
    "Bucket83908E77": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketName": "trivial-backup-02",
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "BucketPolicyE9A3008A": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "Bucket83908E77",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:PutObject",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "User00B015A1",
                    "Arn",
                  ],
                },
              },
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "Bucket83908E77",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "User00B015A1": Object {
      "Properties": Object {
        "UserName": "trivial_backup",
      },
      "Type": "AWS::IAM::User",
    },
  },
}
`;
